### PR TITLE
FIPS on Solaris

### PR DESCRIPTION
**Pull request by tmael**
This PR adds support for building and running FIPS140-2 wolfcrypt test on Solaris.

### CODE REVIEW

File: fips-check.sh [hunk 1]
```diff
@@ -34,13 +34,16 @@ Platform is one of:
...
     wolfrand
+    solaris
 Keep (default off) retains the XXX-fips-test temp dir for inspection.
 
 Example:
     $0 windows keep
 usageText
 }
 
+MAKE=make
```
Comment (ejohnstown) [inline fips-check.sh:45]: Good idea.

File: fips-check.sh [hunk 2]
```diff
@@ -229,6 +232,20 @@ wolfrand)
...
+  FIPS_REPO=git@github.com:wolfssl/fips.git
+  CRYPT_VERSION=WCv4-stable
+  CRYPT_INC_PATH=wolfssl/wolfcrypt
+  CRYPT_SRC_PATH=wolfcrypt/src
+  WC_MODS+=( cmac dh ecc sha3 )
+  RNG_VERSION=WCv4-rng-stable
+  FIPS_SRCS+=( wolfcrypt_first.c wolfcrypt_last.c )
+  FIPS_INCS=( fips.h )
+  FIPS_OPTION=v2
+  CFLAGS_OPTIONS=CFLAGS=-DWOLFSSL_SOLARIS
```
Comment (ejohnstown) [inline fips-check.sh:246]: Please quote this.

    CFLAGS_OPTIONS="CPPFLAGS=-DWOLFSSL_SOLARIS"

File: wolfssl/wolfcrypt/settings.h [hunk 3]
```diff
@@ -209,6 +209,9 @@
 /* Uncomment next line if using RENESAS RX64N */
 /* #define WOLFSSL_RENESAS_RX65N */
 
+/* Uncomment next line if using Solaris OS*/
```
Comment (ejohnstown) [inline wolfssl/wolfcrypt/settings.h:212]: A little after this, you should check for

    #if defined(sun) || defined(__sun)
        #if defined(__SVR4) || defined(__svr4__)
            #define WOLFSSL_SOLARIS
        #endif
    #endif

Comment (ejohnstown) [reply wolfssl/wolfcrypt/settings.h:212]: And then remove it from the fips-check.sh script.

**Comment by ejohnstown** [comment]
I am adding a PR that "fixes" the openssl.test script. It should be checking to see if it should run before doing anything else. And then I fixed the hard failure where head is run with an option that doesn't exist in Solaris's version of head.

File: wolfcrypt/test/test.c [hunk 4]
```diff
@@ -27921,13 +27921,15 @@ int mutex_test(void)
...
+#if !defined(WOLFSSL_SOLARIS)
     if (wc_FreeMutex(&m) != BAD_MUTEX_E)
         return -12703;
+#endif
     if (wc_UnLockMutex(&m) != 0)
         return -12704;
     if (wc_FreeMutex(&m) != 0)
         return -12705;
-#ifndef WOLFSSL_NO_MUTEXLOCK_AFTER_FREE
+#if !defined(WOLFSSL_NO_MUTEXLOCK_AFTER_FREE) && !defined(WOLFSSL_SOLARIS)
```
Comment (toddouska) [inline wolfcrypt/test/test.c:27932]: Why is the SOLARIS define added here?  SOLARIS turns on NO_MUTEXLOCK_AFTER_FREE in settings.

Comment (tmael) [reply wolfcrypt/test/test.c]: @toddouska, Removed.

**Pull request merged**